Source/ConversationV1/Tests/ConversationTests.swift

```diff
@@ -0,0 +1,215 @@
+
+/**
+ * Copyright IBM Corporation 2016
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import XCTest
+import WatsonDeveloperCloud
+import Freddy
+
+class ConversationTests: XCTestCase {
+
+    // MARK: - Parameters and constants
+
+    // the Conversation service
+    var service:     Conversation!
+    var workspaceID: Conversation.WorkspaceID! = "car_demo_1"
+
+    // the initial node and response from Watson
+    let initialMessage = "Hello"
+    let initialResponse = "Welcome to the Watson Car Demo..."
+
+    let secondMessage = "Turn on wipers"
+    let secondResponse = "Your wipers are now on"
+    
+    // invalid parameters for negative tests
+    let invalidWorkspaceID = "9354b734-d5b2-4fd3-bee0-e38adbcab575"
+
+    // timeout for asynchronous completion handlers
+    private let timeout: NSTimeInterval = 30.0
+
+    // MARK: - Conversation Tests
+
+    /// Test the Conversation service by executing each operation with valid parameters.
+    func testConversation() {
```

Comment:
Dialog used to just run one massive test with all of the test cases inside of it. That turned out to be a bad idea for several reasons. Unfortunately, it's challenging to manage the state of the dialog otherwise.

The state can be manually managed with Conversations, though, so we should be able to split it into separate tests (i.e. the `sendInitialMessageNoContext()` function and `sendMessageWithContext()` function can just be the tests).